[18.0][MIG] web_remember_tree_column_width

JasminSForgeFlow opened the pull request:
Standard Migration

Adapt 18.0 new hook code for change/update width of list

@ForgeFlow

frahikLV committed:
[ADD] web_tree_column_width_remember: Remember the tree view columns' widths

Remember the tree columns' widths across sessions, and after filtering, grouping, or reordering.

Commit:
[UPD] Update web_remember_tree_column_width.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

MihranThalhath committed:
[15.0][FIX] web_remember_tree_column_width: use correct module name

this fixes scss file not found error after installing the module

cuongnmtm committed:
[MIG] web_remember_tree_column_width: Migration to 16.0

Commit:
[UPD] Update web_remember_tree_column_width.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

thibaultrey committed:
[FIX] web_remember_tree_column_width : Column resizing with list_button column
Fix an error when retrieving column sizes for lists containing one or more list_button columns.
This error occurs when a list_button column is placed between other columns in the list. This type of column is excluded from the querySelector string used to retrieve column headers, which can cause the index to shift and the column size to change when the list is displayed.

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

cuongnmtm committed:
[IMP] web_remember_tree_column_width: pre-commit auto fixes

cuongnmtm committed:
[MIG] web_remember_tree_column_width: Migration to 17.0

Commit:
[UPD] Update web_remember_tree_column_width.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-17.0/web-17.0-web_remember_tree_column_width
Translate-URL: https://translation.odoo-community.org/projects/web-17-0/web-17-0-web_remember_tree_column_width/

JasminSForgeFlow committed:
[MIG] web_remember_tree_column_width: Migration to 18.0

Co-authored-by: Stefan Rijnhart <stefan@opener.amsterdam>

AaronHForgeFlow commented on web_remember_tree_column_width/static/src/js/list_renderer.esm.js:
```diff
@@ -0,0 +1,23 @@
+/** @odoo-module **/
```
I think this line can be removed safely

JasminSForgeFlow replied:
Updated, Thanks

AaronHForgeFlow commented on web_remember_tree_column_width/readme/CREDITS.md:
```diff
@@ -0,0 +1,6 @@
+- Vauxoo
```
This file can be removed in this version.

JasminSForgeFlow replied:
Updated, Thanks

AaronHForgeFlow reviewed (commented):
Functional review :+1:  Thank you.

Can't do a proper code review, as my knowledge in js is limited. Added a couple of minor comments.

AaronHForgeFlow reviewed (approved):
Thanks! LGTM functional + soft code review :+1:

sanderlienaerts commented:
Can we get this PR merged?

mylbco commented:
Can we get this merged ?

salimChoura1 commented:
Can we get this merged ?

BT-aleonard commented:
Can we merge this, please?

StefanRijnhart reviewed (changes requested):
Mmm, this looks like a modified copy of https://github.com/odoo/odoo/blob/18.0/addons/web/static/src/views/list/column_width_hook.js. Is it documented anywhere that this is the case? Are the changes annotated properly? Is there a more maintainable way of reusing at least parts of the code in that file?

JasminSForgeFlow commented:
> Mmm, this looks like a modified copy of https://github.com/odoo/odoo/blob/18.0/addons/web/static/src/views/list/column_width_hook.js. Is it documented anywhere that this is the case? Are the changes annotated properly? Is there a more maintainable way of reusing at least parts of the code in that file?

that is used as a hook, and it is not possible to inherit it

StefanRijnhart commented:
> > Mmm, this looks like a modified copy of https://github.com/odoo/odoo/blob/18.0/addons/web/static/src/views/list/column_width_hook.js. Is it documented anywhere that this is the case? Are the changes annotated properly? Is there a more maintainable way of reusing at least parts of the code in that file?
> 
> that is used as a hook, and it is not possible to inherit it

But items like `function getWidthSpecs` or the various constants seem unchanged. Can't you at least import these items from the Odoo code? Or is it because they are not exported?

StefanRijnhart commented:
In any case, please document the origin of the code and annotate your changes to it.

JasminSForgeFlow commented:
> > > Mmm, this looks like a modified copy of https://github.com/odoo/odoo/blob/18.0/addons/web/static/src/views/list/column_width_hook.js. Is it documented anywhere that this is the case? Are the changes annotated properly? Is there a more maintainable way of reusing at least parts of the code in that file?
> > 
> > 
> > that is used as a hook, and it is not possible to inherit it
> 
> But items like `function getWidthSpecs` or the various constants seem unchanged. Can't you at least import these items from the Odoo code? Or is it because they are not exported?

Yes, only useMagicColumnWidths is exported, others are using in same file and not exporting so can't import it

JasminSForgeFlow commented:
> In any case, please document the origin of the code and annotate your changes to it.

Added origin, notation for custom code was already there
Thanks

StefanRijnhart commented:
@JasminSForgeFlow Here is a version of the code that does not overwrite the full hook: https://github.com/ForgeFlow/web/pull/4

In that version, I also check if `th` is null in one of the events. It solves an exception that also occurs in this version:

![image](https://github.com/user-attachments/assets/5f37f632-82ea-426b-9b47-8040f1c7943e)

When it happens, the event passes a `th` element that can not be used to retrieve the `td`. The only thing is, when that happens, the modified column width is not stored. Not sure if this is a critical issue.

JasminSForgeFlow commented:
> @JasminSForgeFlow Here is a version of the code that does not overwrite the full hook: [ForgeFlow#4](https://github.com/ForgeFlow/web/pull/4)
> 
> In that version, I also check if `th` is null in one of the events. It solves an exception that also occurs in this version:
> 
> ![image](https://private-user-images.githubusercontent.com/1033124/437387107-5f37f632-82ea-426b-9b47-8040f1c7943e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU4MjU5NDYsIm5iZiI6MTc0NTgyNTY0NiwicGF0aCI6Ii8xMDMzMTI0LzQzNzM4NzEwNy01ZjM3ZjYzMi04MmVhLTQyNmItOWI0Ny04MDQwZjFjNzk0M2UucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDQyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA0MjhUMDczNDA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzAwOWE5YjFkMDIzNzM2YTQ1NjFlMWQ0ZjllZDYzMjNjMWRmYTVlNDZkNDBjMTBlNWUxNTU2ODZhMzI2ZjUyZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.1h7iNgbUuiyUTIn7yR1A7oVP4KGZTdiMp6WvIaokstM)
> 
> When it happens, the event passes a `th` element that can not be used to retrieve the `td`. The only thing is, when that happens, the modified column width is not stored. Not sure if this is a critical issue.

Thanks @StefanRijnhart, I have merged your commit

StefanRijnhart commented:
@JasminSForgeFlow Thanks. You can squash my commit in your migration commit and share credits by mentioning me with `Co-authored-by: ` in the commit message.
Do you experience the same issue with the missing `th` in which case the modified size is not stored? Maybe we should mention it as a known issue.

JasminSForgeFlow commented:
> @JasminSForgeFlow Thanks. You can squash my commit in your migration commit and share credits by mentioning me with `Co-authored-by: ` in the commit message. Do you experience the same issue with the missing `th` in which case the modified size is not stored? Maybe we should mention it as a known issue.

Sure, I'll update commit.

I haven't faced such issue, have you faced an issue with users' last view?

StefanRijnhart commented:
@JasminSForgeFlow Yes, as I indicated I encountered the same error, but uncaught in your version on this PRs runbot. See the screenshot above. I am now catching this condition, but the size cannot be stored in that case.

JasminSForgeFlow commented:
> @JasminSForgeFlow Yes, as I indicated I encountered the same error, but uncaught in your version on this PRs runbot. See the screenshot above. I am now catching this condition, but the size cannot be stored in that case.

But you mentioned that you solved it in your version, didn't you?

StefanRijnhart commented:
@JasminSForgeFlow unfortunately, I did not see how to solve the issue. Like I said,
> I also check if th is null in one of the events. It solves an exception that also occurs in <your> version. When it happens, the event passes a th element that can not be used to retrieve the td. The only thing is, when that happens, the modified column width is not stored. Not sure if this is a critical issue.

JasminSForgeFlow commented:
> @JasminSForgeFlow unfortunately, I did not see how to solve the issue. Like I said,
> 
> > I also check if th is null in one of the events. It solves an exception that also occurs in  version. When it happens, the event passes a th element that can not be used to retrieve the td. The only thing is, when that happens, the modified column width is not stored. Not sure if this is a critical issue.

Okay, I have added it to know issue, and commit is also updated.
Thanks

StefanRijnhart reviewed (approved):
Thanks for taking my suggestions!

TomAlbrechtQuatra reviewed (approved):
Functional & Technical review
Thanks!

StefanRijnhart commented:
@OCA/web-maintainers this one should be ready for merge

etobella reviewed (approved):
/ocabot merge nobump

StefanRijnhart commented:
FYI this module broke 4 days ago due to a change in Odoo. Fixed in https://github.com/OCA/web/pull/3177

Bart-dh commented:
@JasminSForgeFlow This does not work at all in Odoo 18 Enterprise edition. Is it only tested in CE, or is there perhaps something interfering in our database?

etobella commented:
It is tested in CE, not in EE, as some of us don't have access to EE.  Actually, OCA only tests in CE. If you want, you can try to find the reason and make it compatible.

StefanRijnhart commented:
@Bart-dh Using this in EE, but check the known issue discussed above. For us, it works most of the time.

-- edit -- meant to say EE.

AaronHForgeFlow commented:
I tested it in EE and no issues so far.

StefanRijnhart commented:
@Bart-dh I'm suspecting there may be an issue with list views that contain buttons. Buttons may mess up the mapping of fields to list columns. Are there buttons in the views that don't work for you?

Bart-dh commented:
> @Bart-dh I'm suspecting there may be an issue with list views that contain buttons. Buttons may mess up the mapping of fields to list columns. Are there buttons in the views that don't work for you?

Ahh yes, indeed there are buttons! There is the default 'View record' button, but also custom buttons.

StefanRijnhart commented:
I fixed the roadmap issue from this PR in https://github.com/OCA/web/pull/3273, because it becomes quite prevalent in busy treeviews. @Bart-dh you might want to try that version to see if it solves your problems as well.
